packages/flame/lib/src/events/flame_game_mixins/has_tappable_components.dart

```diff
@@ -0,0 +1,203 @@
+import 'package:flame/src/components/mixins/tappable.dart';
+import 'package:flame/src/events/component_mixins/tap_callbacks.dart';
+import 'package:flame/src/events/interfaces/multi_tap_listener.dart';
+import 'package:flame/src/events/messages/tap_cancel_event.dart';
+import 'package:flame/src/events/messages/tap_down_event.dart';
+import 'package:flame/src/events/messages/tap_up_event.dart';
+import 'package:flame/src/game/flame_game.dart';
+import 'package:flutter/gestures.dart';
+import 'package:meta/meta.dart';
+
+/// This mixin allows a [FlameGame] to respond to tap events, and also delivers
+/// those events to components that have the [TapCallbacks] mixin.
+///
+/// The following events are supported by the mixin: [onTapDown], [onTapUp],
+/// [onTapCancel] and [onLongTapDown] -- see their individual descriptions for
+/// more details. There is no "onTap" event though -- use [onTapUp] instead.
+///
+/// Each event handler can be overridden. One scenario when this could be useful
+/// is to check the `event.handled` property after the event has been sent down
+/// the component tree.
+///
+/// === Usage notes ===
+/// - If your game uses components with [TapCallbacks], then this mixin must be
+///   added to the [FlameGame] in order for [TapCallbacks] to work properly.
+/// - If your game also uses [Tappable] components, then add the
+///   [HasLegacyTappables] mixin as well (instead of `HasTappables`).
+/// - If your game has no tappable components, then do not use this mixin.
+///   Instead, consider `MultiTouchTapDetector`.
+mixin HasTappableComponents on FlameGame implements MultiTapListener {
```

Comment:
This is actually a nicer name than `HasTappables` too, so maybe we should deprecate the other one later instead of renaming this one to that name like we discussed.